### PR TITLE
low-code connectors: convert request headers to string before submitting them

### PR DESCRIPTION
**Pull request by girarda**
## What
requests raises an exception if the request headers are not strings
```
raise InvalidHeader(\nrequests.exceptions.InvalidHeader: Header part (1.0) from {'api-version': 1.0} must be of type str or bytes, not <class 'float'>\n", "failure_type": "system_error"}}}
```

## How
* cast request headers to strings before sending them to `HttpStream`

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py [hunk 1]
```diff
@@ -45,7 +45,14 @@ def request_params(
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Mapping[str, Any]:
-        return dict(ChainMap(*[s.request_params() for s in self._stream_slicers]))
+        return dict(
+            ChainMap(
+                *[
+                    s.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py:51]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py [hunk 2]
```diff
@@ -54,7 +61,14 @@ def request_headers(
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Mapping[str, Any]:
-        return dict(ChainMap(*[s.request_headers(stream_state, stream_slice, next_page_token) for s in self._stream_slicers]))
+        return dict(
+            ChainMap(
+                *[
+                    s.request_headers(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py:67]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py [hunk 3]
```diff
@@ -72,7 +93,14 @@ def request_body_json(
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Optional[Mapping]:
-        return dict(ChainMap(*[s.request_body_json(stream_state, stream_slice, next_page_token) for s in self._stream_slicers]))
+        return dict(
+            ChainMap(
+                *[
+                    s.request_body_json(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py:99]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py [hunk 4]
```diff
@@ -63,7 +77,14 @@ def request_body_data(
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Mapping[str, Any]:
-        return dict(ChainMap(*[s.request_body_data(stream_state, stream_slice, next_page_token) for s in self._stream_slicers]))
+        return dict(
+            ChainMap(
+                *[
+                    s.request_body_data(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/cartesian_product_stream_slicer.py:83]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py [hunk 5]
```diff
@@ -127,11 +127,11 @@ def _get_request_options(
         :param paginator_method:
         :return:
         """
-        requester_mapping = requester_method(self.state, stream_slice, next_page_token)
+        requester_mapping = requester_method(stream_state=self.state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py:130]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py [hunk 6]
```diff
@@ -127,11 +127,11 @@ def _get_request_options(
...
         requester_mapping_keys = set(requester_mapping.keys())
-        paginator_mapping = paginator_method(self.state, stream_slice, next_page_token)
+        paginator_mapping = paginator_method(stream_state=self.state, stream_slice=stream_slice, next_page_token=next_page_token)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py:132]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py [hunk 7]
```diff
@@ -127,11 +127,11 @@ def _get_request_options(
...
         paginator_mapping_keys = set(paginator_mapping.keys())
-        stream_slicer_mapping = stream_slicer_method(stream_slice)
+        stream_slicer_mapping = stream_slicer_method(stream_slice=stream_slice)
```
Comment (girarda) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py:134]: pass keyworded arguments

File: airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py [hunk 8]
```diff
@@ -150,13 +150,14 @@ def request_headers(
...
-        return self._get_request_options(
+        request_options = self._get_request_options(
             stream_slice,
             next_page_token,
             self._requester.request_headers,
             self._paginator.request_headers,
             self._stream_slicer.request_headers,
-        )
+        ).items()
+        return {k: str(v) for k, v in request_options}
```
Comment (sherifnada) [inline airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py:160]:
```suggestion
        # HTTP headers are supposed to be strings and a request may fail if they are not. 
        return {k: str(v) for k, v in request_options}
```

**Review by sherifnada — APPROVED** [review]

**Pull request merged**